### ISSUE TITLE
Publish `@tus/client`

### ISSUE THREAD

**Issue by Murderlon** [tus/tus-js-client#522]
Now that [tus-node-server](https://github.com/tus/tus-node-server) is published under the `@tus` npm scope, `tus-js-client` should follow along.

We need to:
- Release a new major version for `tus-js-client` which marks this npm package as deprecated. It will no longer receive updates besides security ones. This also forces people to look up the changelog and they'll find out they need to switch.
- Change package name and publish to npm.

**Comment by Acconut** [comment]
> * Release a new major version for `tus-js-client` which marks this npm package as deprecated.

How does one do this? Using the `npm deprecate` program?

**Comment by Murderlon** [comment]
Indeed! One could argue that a new major is not needed when you deprecate, so that's up to us

**Comment by Acconut** [comment]
Ok, so I think I would do it as the following:

- Release a final patch version: tus-js-client@3.0.2, so that all current patches are available to current users
- Deprecate tus-js-client
- Release @tus/client@3.0.2

Does that sounds the correct plan? I do not think a major release is necessary here.

**Comment by Murderlon** [comment]
Sounds good!

**Comment by Acconut** [comment]
To be honest, I had second thoughts about this. There are a lot of examples, documentation, blog posts and resources covering tus-js-client, so changing the name would be cumbersome for new and existing users without providing much benefit for them. From my own experience, I feel how frustrating it is if the name of a dependency changes that is not frequently interacted with (frankly, I do think that most developers do not use tus-js-client very often, once it has been successfully integrated).

Yes, I do see the benefit of using the @tus scope for tus-node-server, which consists of multiple packages. But for tus-js-client that is not (yet) the case, so I don't see it as useful for this project.

I like the idea of having everything under one umbrella, but when comparing the advantages and downsides, I still think that a name change would cause unnecessary confusion for users. So I am now tending to keeping the package name as tus-js-client.
